src/main/java/aegis/Aegis.java

```diff
@@ -0,0 +1,282 @@
+                break;
+            case "delete":
+                deleteTask(taskList, input);
+                break;
+            default:
+                throw new AegisException(" Your command has not been authorized ");
+            }
+        } catch (AegisException e) {
+            System.out.printf(" Anomaly detected: %s%n", e.getMessage());
+        }
+    }
+
+    private static void displayTaskList(ArrayList<Task> taskList) {
+        System.out.println(" I could do much more than checking your list, but, if you wish:");
+        for (int i = 0; i < taskList.size(); i++) {
+            System.out.printf("   %d. %s%n", i + 1, taskList.get(i));
+        }
+    }
+
+    private static void markTaskAsDone(ArrayList<Task> taskList, String input) {
+        try {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+            if (taskIndex < 0 || taskIndex >= taskList.size()) {
+                throw new AegisException("Invalid task number");
+            }
+            taskList.get(taskIndex).markAsDone();
+            saveTasksToFile(taskList);
+            System.out.printf(" I've marked this task as done:%n   %s%n", taskList.get(taskIndex));
+        } catch (AegisException | NumberFormatException e) {
+            System.out.printf(" Anomaly detected: [%s]%n", e.getMessage());
+        }
+    }
+
+    private static void unmarkTaskAsDone(ArrayList<Task> taskList, String input) {
+        try {
+            int taskIndex = Integer.parseInt(input.split(" ")[1]) - 1;
+            if (taskIndex < 0 || taskIndex >= taskList.size()) {
+                throw new AegisException("Invalid task number");
+            }
+            taskList.get(taskIndex).unmarkAsDone();
+            saveTasksToFile(taskList); // Save after unmarking task
+            System.out.printf(" I've marked this task as not done yet:%n   %s%n", taskList.get(taskIndex));
+        } catch (AegisException | NumberFormatException e) {
+            System.out.printf(" Anomaly detected: [%s]%n", e.getMessage());
+        }
+    }
+
+    private static void addDeadline(ArrayList<Task> taskList, String input) {
+        try {
+            if (!input.contains(" /by ")) {
+                throw new AegisException("Invalid syntax: you need to include [/by] parameter");
+            }
+
+            String[] parts = input.split(" /by ", 2);
+            String description = parts[0].replaceFirst("deadline ", "").trim();
+            String byTime = parts[1].trim();
+
+            Deadline newDeadline = new Deadline(description, byTime);
+            taskList.add(newDeadline);
+            saveTasksToFile(taskList);
+            System.out.printf(" New deadline added: %n   %s%n", newDeadline);
+            System.out.printf(" %d tasks needed to be done. Let me assist you!%n", taskList.size());
+        } catch (AegisException e) {
+            System.out.printf(" Anomaly detected: [%s]%n", e.getMessage());
+        }
```

Comment:
The code here can be extracted as well as the same pattern is used in several other methods as well.